keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/InteractiveNotificationCallbackController.kt

```diff
@@ -18,29 +21,31 @@ import org.springframework.web.bind.annotation.RestController
 @RestController
 @RequestMapping(path = ["/notifications/callback"])
 class InteractiveNotificationCallbackController(
-  private val repository: KeelRepository
+  private val repository: KeelRepository,
+  private val authorizationSupport: AuthorizationSupport
 ) {
   private val log by lazy { LoggerFactory.getLogger(javaClass) }
 
   @PostMapping(
     consumes = [MediaType.APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE],
     produces = [MediaType.APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE]
   )
-  // TODO: This should be validated against write access to a service account. Service accounts should
-  //  become a top-level property of either delivery configs or environments.
-  //
-  // TODO(lfp): Related to the above, we'll need an additional authentication method for interactive constraint
+  // TODO(lfp): We might need an additional authentication method for interactive constraint
   //  approval outside of the Spinnaker UI, e.g. in Slack, since X-SPINNAKER-USER will be extracted from the Slack
   //  message and not provided by the UI. My plan is to include an OTP in the callback URL. Note that echo
   //  does token/signature verification of messages, so there's relatively tight security there, but we still trust
-  //  the e-mail address without a Spinnaker user having authenticated here.
+  //  the e-mail address without a user having actually authenticated with Spinnaker.
   fun handleInteractionCallback(
     @RequestHeader("X-SPINNAKER-USER") user: String,
     @RequestBody callback: EchoNotification.InteractiveActionCallback
   ) {
     val currentState = repository.getConstraintStateById(parseUID(callback.messageId))
       ?: throw InvalidConstraintException("constraint@callbackId=${callback.messageId}", "constraint not found")
 
+    if (!authorizationSupport.userCan(Action.WRITE, Entity.DELIVERY_CONFIG, currentState.deliveryConfigName)) {
```

Comment:
Unfortunately, for this API, since it must be kept generic to comply with the contract for interactive notifications, I have to look up the constraint state from the database to then check permissions. But, it's an example showing you can use `AuthorizationSupport` directly instead of the annotations, if needed.